File: libs/document-processor/src/repo/core/schema/Operation.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId, blueNodeField } from '@blue-labs/language';

export const OperationSchema = withTypeBlueId(blueIds['Operation'])(
  z.object({
    request: blueNodeField().optional(),
    description: z.string().optional(),
    channel: z.string().optional(),
  })
);

export type Operation = z.infer<typeof OperationSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/Operation.java
package blue.language.types.conversation;

import blue.language.model.Node;
import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

@TypeAlias("Conversation/Operation")
@TypeBlueId("BoAiqVUZv9Fum3wFqaX2JnQMBHJLxJSo2V9U2UBmCfsC")
public class Operation {
    public Node request;
    public String channel;

    public Operation request(Node request) {
        this.request = request;
        return this;
    }

    public Operation channel(String channel) {
        this.channel = channel;
        return this;
    }
}
